packages/caver-wallet/src/keyring/multipleKeyring.js

```diff
@@ -0,0 +1,213 @@
+        for (let i = 0; i < this.keys.length; i++) {
+            publicKeys.push(this.keys[i].getPublicKey())
+        }
+        return publicKeys
+    }
+
+    /**
+     * returns a copied multipleKeyring instance
+     *
+     * @return {MultipleKeyring}
+     */
+    copy() {
+        return new MultipleKeyring(this.address, this.keys)
+    }
+
+    /**
+     * signs with transactionHash with key and returns signature(s).
+     *
+     * @param {string} transactionHash The hash of transaction.
+     * @param {string|number} chainId The chainId specific to the network.
+     * @param {number} role A number indicating the role of the key. You can use `caver.wallet.keyring.role`.
+     * @param {number} [index] The index of the key to be used. If index is undefined, all private keys in keyring will be used.
+     * @return {Array.<string>|Array.<Array.<string>>}
+     */
+    sign(transactionHash, chainId, role, index) {
+        validateForSigning(transactionHash, chainId)
+
+        const keys = this.getKeyByRole(role)
+
+        if (index !== undefined) {
+            validateIndexWithKeys(index, keys.length)
+            return keys[index].sign(transactionHash, chainId)
+        }
+
+        const signatures = []
+        for (const k of keys) {
+            signatures.push(k.sign(transactionHash, chainId))
+        }
+        return signatures
+    }
+
+    /**
+     * signs with hashed message and returns result object that includes `signatures`, `message` and `messageHash`
+     *
+     * @param {string} message The message string to sign.
+     * @param {number} role A number indicating the role of the key. You can use `caver.wallet.keyring.role`.
+     * @param {number} [index] The index of the key to be used.
+     * @return {object}
+     */
+    signMessage(message, role, index) {
+        if (role === undefined) throw new Error(`role should be defined for signMessage. Please use 'caver.wallet.keyring.role'.`)
+        const messageHash = utils.hashMessage(message)
+
+        const keys = this.getKeyByRole(role)
+
+        const signatures = []
+        if (index !== undefined) {
+            validateIndexWithKeys(index, keys.length)
+            signatures.push(keys[index].signMessage(messageHash))
+        } else {
+            for (const k of keys) {
+                signatures.push(k.signMessage(messageHash))
+            }
+        }
+        return {
+            messageHash,
+            signatures,
+            message,
+        }
+    }
+
+    /**
+     * returns keys by role.If the key of the role passed as parameter is empty, the default key is returned.
+     *
+     * @param {number} role A number indicating the role of the key. You can use `caver.wallet.keyring.role`.
+     * @return {Array.<PrivateKey>}
+     */
+    getKeyByRole(role) {
+        if (role === undefined) throw new Error(`role should be defined.`)
+        if (role >= KEY_ROLE.roleLast) throw new Error(`Invalid role number: ${role}`)
```

Comment:
Don't we need to check negative indexes?

Reply:
I added "role < 0" check in if condition :) Thank you!